pkg/e2e/framework/nodes.go

```diff
@@ -0,0 +1,39 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetWorkerNodes returns all nodes with the nodeWorkerRoleLabel label
+func GetWorkerNodes(client runtimeclient.Client) ([]corev1.Node, error) {
+	listOptions := runtimeclient.ListOptions{
+		Namespace: TestContext.MachineApiNamespace,
+	}
+	listOptions.SetLabelSelector(fmt.Sprintf("%s=", NodeWorkerRoleLabel))
+	workerNodes := &corev1.NodeList{}
+	err := client.List(context.TODO(), &listOptions, workerNodes)
+	if err != nil {
+		return nil, err
+	}
+	return workerNodes.Items, nil
+}
+
+// GetWorkerReadyNodes returns all ready nodes with the nodeWorkerRoleLabel label
+func GetWorkerReadyNodes(client runtimeclient.Client) ([]corev1.Node, error) {
```

Comment:
Can you rename the function to `FilterReadyNodes` and pass a list of nodes instead? We already have a use case where we verify all nodes (including master nodes) are ready.

Reply:
Done
Can you point me where can I replace the code with the `FilterReadyNodes`?